src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSMultiBatchSink.java

```diff
@@ -161,11 +188,18 @@ private void configureMultiSinkWithSchema(BatchSinkContext context,
       }
       String name = key.substring(TABLE_PREFIX.length());
       Schema schema = Schema.parseJson(argument.getValue());
+
       // TODO: (CDAP-14600) pass in schema as an argument instead of using macros and setting arguments
       // add better platform support to allow passing in arguments when instantiating a plugin
       context.getArguments().set(SCHEMA_MACRO, schema.toString());
-      ValidatingOutputFormat validatingOutputFormat = context.newPluginInstance(FORMAT_PLUGIN_ID);
 
+      String macroFormat = StringUtils.EMPTY;
+      if (config.getRawProperties().getProperties().get("format").contains("{")) {
```

Comment:
At `prepareRun` time macros are expected to be resolved so need to check this condition.

Reply:
@itsankit-google Sorry i didn't understand this.
Btw i verified that this piece of code has been added to prepareRun flow itself.

Reply:
So, that's what my meant if this is added in `prepareRun`. We don't need to check if config is macro or not. The framework is defined in such a way, it is expected to be non-macro at this time.

Reply:
okays, got it, in any case the format value has been resolved at this stage. Thanks done